packages/dart_firebase_admin/lib/src/messaging/fmc_exception.dart

```diff
@@ -59,83 +59,167 @@ const _topicMgtServerToClientCode = {
+  /// You should use one of those sender IDs when sending messages to the client app.
+  /// If you switch to a different sender, the existing registration tokens won't work.
+  ///
   senderIdMismatch(
     code: 'SENDER_ID_MISMATCH',
-    'The authenticated sender ID is different from the sender ID for the registration token.',
+    message:
+        'The authenticated sender ID is different from the sender ID for the registration token.',
+    httpStatusCode: 403,
   ),
 
+  ///
+  /// A message targeted to an iOS device or a web push registration could not be sent.
+  /// Check the validity of your development and production credentials.
+  ///
   thirdPartyAuthError(
     code: 'THIRD_PARTY_AUTH_ERROR',
-    'APNs certificate or web push auth key was invalid or missing.',
+    message: 'APNs certificate or web push auth key was invalid or missing.',
+    httpStatusCode: 401,
   ),
 
+  ///
+  /// The server couldn't process the request in time. Retry the same request, but you must:
+  ///
+  /// - Honor the Retry-After header if it is included in the response from the FCM Connection Server.
+  /// - Implement exponential back-off in your retry mechanism.
+  /// (e.g. if you waited one second before the first retry, wait at least two
+  /// second before the next one, then 4 seconds and so on). If you're sending
+  /// multiple messages, delay each one independently by an additional random
+  /// amount to avoid issuing a new request for all messages at the same time.
+  /// Senders that cause problems risk being denylisted.
+  ///
   unavailable(
     code: 'UNAVAILABLE',
-    'Cloud Messaging service is temporarily unavailable.',
+    message: 'Cloud Messaging service is temporarily unavailable.',
+    httpStatusCode: 503,
   ),
 
+  ///
+  /// This error can be caused by missing registration tokens, or unregistered tokens.
+  ///
+  /// Missing Registration: If the message's target is a token value, check that
+  /// the request contains a registration token.
+  ///
+  /// Not registered: An existing registration token may cease to be valid in
+  /// a number of scenarios, including:
+  /// - If the client app unregisters with FCM.
+  /// - If the client app is automatically unregistered, which can happen if the
+  /// user uninstalls the application. For example, on iOS, if the APNS Feedback Service
+  /// reported the APNS token as invalid.
+  /// - If the registration token expires (for example, Google might decide to
+  /// refresh registration tokens, or the APNS token has expired for iOS devices).
+  /// - If the client app is updated but the new version is not configured to
+  /// receive messages.
+  ///
+  /// For all these cases, remove this registration token from the app server and stop using it to send messages.
   unregistered(
     code: 'UNREGISTERED',
-    'App instance was unregistered from FCM. '
-    'This usually means that the token used is no longer valid and a new one must be used.',
-  ),
-
-  authenticationError(
```

Comment:
Those codes are official errors which you'd find in the JS implementation too. I don't think removing them makes sense